webapp/src/views/projects/translations/TranslationHeader/TranslationControlsCompact.tsx

```diff
@@ -0,0 +1,317 @@
+  const view = useTranslationsSelector((v) => v.view);
+  const selectedLanguages = useTranslationsSelector((c) => c.selectedLanguages);
+  const [anchorFiltersEl, setAnchorFiltersEl] =
+    useState<HTMLButtonElement | null>(null);
+  const [anchorLanguagesEl, setAnchorLanguagesEl] =
+    useState<HTMLButtonElement | null>(null);
+
+  const handleSearchChange = (value: string) => {
+    dispatch({ type: 'SET_SEARCH', payload: value });
+  };
+
+  const filtersContent = useFiltersContent();
+  const activeFilters = useActiveFilters();
+
+  const handleLanguageChange = (languages: string[]) => {
+    dispatch({
+      type: 'SELECT_LANGUAGES',
+      payload: languages,
+    });
+  };
+
+  const handleViewChange = (val: ViewMode) => {
+    dispatch({ type: 'CHANGE_VIEW', payload: val });
+  };
+
+  const handleDelete = () => {
+    dispatch({ type: 'DELETE_TRANSLATIONS' });
+  };
+
+  const handleClearFilters = () => {
+    dispatch({ type: 'SET_FILTERS', payload: {} });
+  };
+
+  const handleAddTranslation = () => {
+    onDialogOpen();
+  };
+
+  const trigger = useTopBarHidden();
+
+  return (
+    <>
+      <div
+        className={classes.controls}
+        style={{
+          transform: trigger ? 'translate(0px, -50px)' : 'translate(0px, 0px)',
+        }}
+      >
+        {searchOpen ? (
+          <div className={classes.searchSpaced}>
+            <TranslationsSearchField
+              value={search || ''}
+              onSearchChange={handleSearchChange}
+              className={classes.search}
+              label={null}
+              variant="outlined"
+              placeholder={t('standard_search_label')}
+              style={{
+                height: 35,
+                maxWidth: 'unset',
+                width: '100%',
+              }}
+            />
+            <IconButton
+              size="small"
+              className={classes.iconButton}
+              onClick={() => setSearchOpen(false)}
+            >
+              <Clear />
+            </IconButton>
+          </div>
+        ) : (
+          <>
+            <div className={classes.spaced}>
+              {selection.length > 0 && (
+                <IconButton
+                  className={classes.deleteButton}
+                  onClick={handleDelete}
+                  data-cy="translations-delete-button"
+                >
+                  <Delete />
+                </IconButton>
+              )}
+              <Badge color="primary" badgeContent={search.length} variant="dot">
+                <div className={classes.buttonWrapper}>
+                  <IconButton
+                    size="small"
+                    className={classes.iconButton}
+                    onClick={() => setSearchOpen(true)}
+                  >
+                    <Search />
+                  </IconButton>
+                </div>
+              </Badge>
+
+              <Badge color="primary" badgeContent={activeFilters?.length}>
```

Comment:
Maybe extract the menus to separate component?

Reply:
Ok